### PR TITLE
Define tag build

### PR DESCRIPTION
**Pull request by giorgiosironi**
Highly experimental. 

Will be running at https://alfred.elifesciences.org/job/pull-requests-projects/job/sciencebeam-utils/view/tags/ in response to a new tag being pushed.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -48,14 +47,11 @@ elifeLibrary {
...
-        elifeMainlineOnly {
+        elifeTagOnly { tag ->
             stage 'Push release', {
-                isNew = sh(script: "git tag | grep v${candidateVersion}", returnStatus: true) != 0
-                if (isNew) {
-                    sh "IMAGE_TAG=${commit} " +
-                        "docker-compose -f docker-compose.yml -f docker-compose.ci.yml run " +
-                        "sciencebeam-utils-py2 twine upload dist/*"
-                }
+                sh "IMAGE_TAG=${commit} " +
```
Comment (giorgiosironi) [inline Jenkinsfile]: doesn't seem to need a version input, because it's still stored in the code

File: Jenkinsfile [hunk 2]
```diff
@@ -64,9 +60,6 @@ elifeLibrary {
     elifeMainlineOnly {
         stage 'Merge to master', {
             elifeGitMoveToBranch commit, 'master'
-            if (isNew) {
-                sh "git tag v${candidateVersion} && git push origin v${candidateVersion}"
```
Comment (giorgiosironi) [inline Jenkinsfile]: no need to tag again, since it's the tag triggering any build

**Comment by giorgiosironi** [comment]
See also https://github.com/elifesciences/sciencebeam/pull/126 which is in a limbo.

File: Jenkinsfile (hunk 1 above)
Comment (de-code) [reply Jenkinsfile]: should we maybe generate a versions file that is read by `__init__.py`?

Comment (giorgiosironi) [reply Jenkinsfile]: Seems to still potentially contradict https://github.com/elifesciences/tech-team/pull/17 but it's reasonable for an application to know its own version number in some way. What it is used for?

Comment (de-code) [reply Jenkinsfile]: It's an optional convention to provide the version number, specified by [PEP 396](https://www.python.org/dev/peps/pep-0396/).
I think it's still in compliance with the ADR if the version number is derived from the tag. It doesn't necessarily need to appear in GitHub, as long as it's part of the release published to pypi.

Comment (giorgiosironi) [reply Jenkinsfile]: Seems like the best of both worlds, no forgotten version numbers committed in `master` and the capability to introspect the version from Python when installing the library.
How do we add it? Seems part of the `Dockerfile` with a build argument being passed?

Comment (giorgiosironi) [reply Jenkinsfile]: Implemented with a `sed` in the `Dockerfile`.

File: .env [hunk 3]
```diff
@@ -1,2 +1,3 @@
 COMPOSE_PROJECT_NAME=sciencebeam-utils
 IMAGE_TAG=develop
+VERSION=develop
```
Comment (giorgiosironi) [inline .env]: feel free to change to another default value

**Comment by de-code** [comment]
Do think this might be ready to merge and build a release for?

**Comment by giorgiosironi** [comment]
No blockers from me.

**Pull request merged**